src/flowchem/devices/knauer/knauer_autosampler_component.py

```diff
@@ -0,0 +1,314 @@
+            if success:
+                logger.info("Needle moved successfully to row: {row}, column: {column} on plate: {plate}")
+            return
+        else:
+            raise NotImplementedError
+
+    async def set_z_position(self, direction: str = "") -> None:
+        """
+        Move the 3D gantry along the Z axis.
+
+        Argument:
+            direction (str):
+            DOWN
+            UP
+        """
+        if await self.hw_device.get_status() == "NEEDLE_RUNNING":
+            logger.warning("Needle already moving!")
+
+        success = await self.hw_device._move_needle_vertical(move_to=direction)
+        if success:
+            logger.info("Needle moved successfully to {direction} direction.")
+
+
+    #ToDo
+    async def get_position(self) -> tuple:
+        """
+        Get the current position of the 3D gantry.
+        If needle is above the plates:
+            Returns a tuple (row, column, UP/DOWN).
+        Else:
+            Returns a string representing one of the predefined positions:
+            position (str):
+                        WASH
+                        WASTE
+                        EXCHANGE
+                        TRANSPORT
+        """
+        ...
+
+
+class AutosamplerPump(SyringePump):
+    """
+    Control a Knauer Autosampler syringe pump.
+
+    Attributes:
+        hw_device (KnauerAutosampler): The hardware device for the Knauer Autosampler syringe pump..
+    """
+
+    hw_device: KnauerAutosampler  # for typing's sake
+
+    def __init__(self, name: str, hw_device: KnauerAutosampler) -> None:
+        """Initialize component."""
+        super().__init__(name, hw_device)
+
+    async def infuse(self, rate: str = None, volume: str = None) -> bool:  # type: ignore
+        """
+        Dispense with built in syringe.
+        Args:
+            volume: volume to dispense in mL
+
+        Returns: None
+        """
+        if volume is None:
+            volume = "0 mL"
+            logger.warning(f"the volume to infuse is not provided. set to 0 ml")
+        parsed_volume = ureg.Quantity(volume)
+        success = await self.hw_device.dispense(volume=parsed_volume.m_as("mL"))
+        if success:
+            logger.info(f"Syringe pump successfully infused {volume} ml")
+
+    async def withdraw(self, rate: str = None, volume: str = None) -> bool:  # type: ignore
+        """
+        Aspirate with built in syringe.
+        Args:
+            volume: volume to aspirate in mL
+
+        Returns: None
+        """
+        if volume is None:
+            volume = self.hw_device.syringe_volume
+            logger.warning(f"the volume to withdraw is not provided. set to {self.hw_device.syringe_volume}")
+        parsed_volume = ureg.Quantity(volume)
+        success = await self.hw_device.aspirate(volume=parsed_volume.m_as("mL"))
+        if success:
+            logger.info(f"Syringe pump successfully withdrew {volume} ml")
+
+    async def stop(self):  # type: ignore
```

Comment:
Since these methods are not implemented, apparently, they should not overwrite parent method. They are not available so do not pretend to the user that those exist